cli/pcluster/networking/network_resource_creation.py

```diff
@@ -0,0 +1,235 @@
+            SubnetRouteTableAssociation(
+                "SubnetRouteTableAssociation", SubnetId=Ref(subnet), RouteTableId=Ref(route_table)
+            )
+        )
+        return route_table
+
+    def _create_nat(self, compute_subnet, condition, master_subnet, vpc):
+        nat_eip = self.template.add_resource(EIP("NatEip", Domain=Ref(vpc), Condition=condition))
+        nat = self.template.add_resource(
+            NatGateway(
+                "Nat",
+                AllocationId=GetAtt(nat_eip, "AllocationId"),
+                SubnetId=Ref(master_subnet),
+                Condition=condition,
+                Tags=CFNSubnetsGenerator._add_name("NAT"),
+            )
+        )
+        private_route_table = self.template.add_resource(
+            RouteTable(
+                "PrivateRouteTable",
+                VpcId=Ref(vpc),
+                Tags=CFNSubnetsGenerator._add_name("Private-Route-Table"),
+                Condition=condition,
+            )
+        )
+        self.template.add_resource(
+            Route(
+                "NatRoute",
+                RouteTableId=Ref(private_route_table),
+                DestinationCidrBlock="0.0.0.0/0",
+                NatGatewayId=Ref(nat),
+                Condition=condition,
+            )
+        )
+        self.template.add_resource(
+            SubnetRouteTableAssociation(
+                "SubnetRouteTableAssociation2",
+                SubnetId=Ref(compute_subnet),
+                RouteTableId=Ref(private_route_table),
+                Condition=condition,
+            )
+        )
+
+    def create(self, path):
+        """Create the yaml file given the path name."""
+        vpc = self._add_parameter(name="VpcId", description="The vpc id", expected_input_type="String")
+
+        # Parameters
+        master_subnet_cidr = self._add_parameter(
+            name="MasterSubnetCidr",
+            description="CIDR describing the size of the master subnet",
+            expected_input_type="String",
+        )
+        compute_subnet_cidr = self._add_parameter(
+            name="ComputeSubnetCidr",
+            description="(Optional) CIDR describing the size of the compute subnet",
+            expected_input_type="String",
+        )
+        availability_zone = self._add_parameter(
+            name="AvailabilityZone",
+            description="(Optional) The zone in which you want to create your subnet(s)",
+            expected_input_type="String",
+        )
+
+        # Conditions
+        is_compute_cidr_set = self.template.add_condition("IsComputeCIDRSet", Not(Equals(Ref(compute_subnet_cidr), "")))
+
+        # Always created Resources
+        master_subnet = self.template.add_resource(
+            Subnet(
+                "MasterSubnet",
+                CidrBlock=Ref(master_subnet_cidr),
+                VpcId=Ref(vpc),
+                AvailabilityZone=Ref(availability_zone),
+                Tags=CFNSubnetsGenerator._add_name("Master-Subnet"),
```

Comment:
let's uniform the convention for the names: if we do aws-parallelcluster then let's add -master-subnet otherwise we need to capitalize the rest. Same applies to all the Name tags